File: src/cmd.py
import subprocess

target_dir = "/"
status = subprocess.run(["ls", target_dir],
                        stdout=subprocess.PIPE,
                        stderr=subprocess.PIPE)

if status.returncode:
    msg = status.stderr.decode()
    raise Exception(msg)
else:
    list_files = status.stdout.decode().split()
    print(list_files)

File: src/cmd.rs
use std::process::Command;

fn main() {

    let cmd = Command::new("ls")
            .arg("/")
            .stdout(std::process::Stdio::piped())
            .output()
            .expect("Failed to invoke command");

    if cmd.status.code() == Some(0) { // cmd.status.success() also available
        let stdout = String::from_utf8(cmd.stdout).unwrap();
        let list_files: Vec<&str> = stdout.split("\n").collect();
        println!("{:?}", list_files);
    } else {
        let stderr = String::from_utf8(cmd.stderr).unwrap();
        println!("Error: {}", stderr.trim());
    }

}
